### DISCUSSION TITLE
How to make tooltip disappear when typing

### DISCUSSION THREAD

**Discussion by CauePizzol** [AvaloniaUI/Avalonia#8833] [Q&A] [answered]
Hi all,
I have a tooltip in a text box and I would like for it to disappear after I start typing on it. Is it normal that it doesn't?
my axaml looks something like this:
```
<Border ToolTip.Tip="My ToolTip">
	<TextBox />
</Border>
```
I would also like for it to disappear as soon as I move the pointer. Currently, if the pointer moves, but stays within the text box, the tooltip will stay in the same position.

My first attempt was to listen to the event of the tooltip opening/closing and setting IsOpen manually. But I was never able to trigger the event upon open or closing.

Thanks!

**Comment by timunie** [comment, marked as answer]
ToolTip has attached property IsOpen. I think you can try to set it to false in your code behind.

**Reply by timunie** [reply]
> ToolTip has attached property IsOpen. I think you can try to set it to false in your code behind.

Update: Here is a working code: 

```cs
private void TextBox_GotFocus(object sender, GotFocusEventArgs e)
{
    if (sender is TextBox textBox)
    {
        // Close the tooltip 
        ToolTip.SetIsOpen(textBox.FindLogicalAncestorOfType<Border>(), false);
    }
}
```

**Reply by CauePizzol** [reply]
I paired this with "KeyDown" event and it works like a charm! Thanks.
